### ISSUE TITLE
Configuration/permission issues for rootless podman containers

### ISSUE THREAD

**Issue by c-herz** [b-m-f/wg-pod#6]
Hello,

I'm facing an issue integrating wg-pod with a rootless Podman container setup, where containers are run by the `podman` user, and everything is managed via Quadlet/systemd. 
My quadlet file looks like so:

```
[Unit]
Description=qbittorrent

[Container]
Image=docker.io/qbittorrentofficial/qbittorrent-nox:latest
AutoUpdate=registry
PublishPort=8088:8088/tcp
Volume=qbt_config:/config:Z
Volume=/orlop/media/downloads:/downloads:z
Environment=QBT_LEGAL_NOTICE=confirm
Environment=QBT_WEBUI_PORT=8088

[Service]
ExecStartPost=/usr/local/bin/wg-pod systemd-qbittorrent /home/podman/.local/wg0.conf
[Install]
# Start by default on boot
WantedBy=default.target
```

The problem arises because running wg-pod as a rootless user results in `RTNETLINK answers: Operation not permitted.` I have given wg-pod `CAP_NET_ADMIN` and this did not resolve the issue. Conversely, running wg-pod as root from the quadlet file, using a limited scope sudo permission granted to the podman user, the rootful wg-pod instance cannot see the rootless Podman container, resulting in no such object errors. Are there ways to allow wg-pod to operate with elevated privileges, i.e. network configuration, within a rootless environment? Or are there any ways to bridge container visibility between root and rootless Podman contexts?

I'm running Rocky Linux 9.4 on an x86 host machine. If you need any more data or configuration information from me, I'd be happy to provide it.

**Comment by b-m-f** [comment]
I think this will need a new flag to specify the user that the container is running in.

The container namespace is retrieved here: https://github.com/b-m-f/wg-pod/blob/c4c9e371c37452ae96bf29a9e199c4e4bef184a1/pkg/podman/podman.go#L40

and could be augmented to search in the containers of the user.

I do not remember the exact rootless network setup right now, but if the container has its own network namespace wg-pod should still work.

**Comment by c-herz** [comment]
So I tentatively have solved it. I forked the repo and did a bit of tinkering: I added a -u (url) flag for the join command, corresponding to the url (path) of the libpod unix socket that can be exposed with the command "podman system service -t 0" by the rootless podman user. Podman lets you connect to any running podman instance with the [libpod API](https://docs.podman.io/en/stable/_static/api.html?version=v4.9). This lets wg-pod, running as root, inspect the rootless podman user's containers and namespaces. Since the netns created rootlessly by slirp4netns doesn't get put under the default /var/run/netns, rather instead into /run/user/[UID]/netns, wg-pod doens't see it, even as root. To remedy this, I added a function to podman.go that uses the aforementioned Podman API to identify the named container, find its PID, and then attach to the created netns with`ip netns attach [NETNS_NAME] [CONTAINER_PID]`. This makes it visible to the system just like any other netns created with `ip netns`, and allows wg-pod to then move the wireguard interface into it, and everything works! So to recap, rootless container hooks wg-pod, which runs as root with sudo, links the rootless netns so it's visible to the system, and then everything works out. It's https://github.com/c-herz/wg-pod/tree/add-url-tag

I'll still need to document the things that I added and probably rename the URL flag to something more appropriate, like `--socket-path` or something, but I can open a PR once I finish. Thanks again for this great project!
